packages/diagrams/lib/src/expanded.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+
+import 'diagram_step.dart';
+
+const String _column = 'expanded_column';
+const String _row = 'expanded_row';
+
+class ExpandedDiagram extends StatelessWidget implements DiagramMetadata {
+  const ExpandedDiagram(this.name);
+
+  @override
+  final String name;
+
+  @override
+  Widget build(BuildContext context) {
+    Widget returnWidget;
+
+    switch (name) {
+      case _column:
+        returnWidget = Scaffold(
+          appBar: AppBar(
+            title: const Text('Expanded Column Sample'),
+          ),
+          body: Center(
+            child: Column(
+              children: <Widget>[
+                Container(
+                  color: Colors.green,
+                  height: 100,
+                ),
+                Expanded(
+                  child: Container(
+                    color: Colors.yellow,
+                    child: const Center(child: Text('Expanded')),
+                  ),
+                ),
+                Container(
+                  color: Colors.green,
+                  height: 100,
+                ),
+              ],
+            ),
+          ),
+        );
+        break;
+      case _row:
+        returnWidget = Scaffold(
+          appBar: AppBar(
+            title: const Text('Expanded Row Sample'),
+          ),
+          body: Center(
+            child: Row(
+              children: <Widget>[
+                Expanded(
+                  flex: 2,
+                  child: Container(
+                    color: Colors.amber,
+                    height: 100,
+                    child: const Text('Expanded(\n  flex: 2\n)'),
```

Comment:
Let's see if we can center, or create a label widget.

Reply:
Yah, this is slightly trickier -- I originally split the text on newlines because I felt it looks a bit awkward when the `Expanded(flex: 1)` text didn't fit on a single line. Here I've centered the text and removed the `\n` newlines and this is the result:

![expanded_row](https://user-images.githubusercontent.com/126604/62375618-fedce500-b53e-11e9-872c-b7187b0307fd.png)

What do you think of this version? I didn't want to nest too many Widgets in here to keep the sample in the API docs simple and didn't want folks copy-pasting the sample to get a different output than what the image shows. Any feedback / ideas appreciated!

Reply:
I see what you mean. 

I would suggest trying to just include 'flex: n' and see how that looks, or make use of the label widget https://github.com/flutter/assets-for-api-docs/blob/3333f4a2832912c1e80220e5e053290f6c764374/packages/diagrams/lib/src/utils.dart#L37 that `utils.dart` provides for this type of thing.

We don't have to worry about the sample in the API docs getting too complicated, because we don't need to include things like labels in the actual snippet. We can use the diagrams to point out or illustrate the different elements and features, but not necessarily/explicitly include those visual aids in the actual snippet. Does that make sense?